app/components/codelist-form.js

```diff
@@ -30,7 +30,10 @@ export default class CodelistFormComponent extends Component {
 
   @action
   async didInsert() {
-    this.options = await this.args.codelist.concepts;
+    const concepts = (await this.args.codelist.concepts).toArray();
+    this.options = concepts.sort((a, b) =>
+      a.createdOn < b.createdOn ? 1 : -1
+    );
```

Comment:
The concepts relation of the codelist is converted to an array here and sorted. When new concepts (options) get added to that array. It isnt stored in the codelist model instance anymore. This is why the new codelist options are not added to the codelist itself.